File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        # if all 1 or if only one 0 return len - 1
        if sum(nums) >= len(nums) - 1:
            return len(nums) - 1
        
        l = 0
        r = 0
        max_len = 0
        zero_counter = 0
        
        while r < len(nums):
            if nums[r] == 0:
                zero_counter += 1
            
            while zero_counter > 1:
                if nums[l] == 0:
                    zero_counter -= 1
                l +=1
                
            max_len = max(max_len, r - l)
            r +=1
            
        return max_len
            
        
        
        
        
        

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int l = 0; // left pointer
        int r = 0; // right pointer
        int zeroCounter = 0; // no more than 1 zero allowed
        int maxLen = 0;
        while (r < nums.length) {
            if (nums[r] == 0) {
                zeroCounter++;
            }
            while (zeroCounter > 1) {
                // move left pointer
                if (nums[l] == 0) {
                    zeroCounter--;
                }
                l++;
            }
            maxLen = Math.max(maxLen, r - l);
            r++;
        }
        return maxLen;
    }
}